ItemReader/Writer can not be declared using producer methods

When I declare an ItemReader or ItemWriter using `@Produces` factory method:
```
    @Produces
    @Named("myReader")
    @Dependent
    CustomItemReader myReader() {
        return new ...;
    }
```
Quarkus JBeret is unable to create the Read/Writer instance:
```
Caused by: java.lang.IllegalArgumentException: Type class org.acme.MyJobProvider is not a bean type of PRODUCER_METHOD bean [class=org.acme.MyJobProvider, id=005ac9d0e79901ee6b2aaacef5e95b54cc3a17f4]; its bean types are: [class java.lang.Object, interface jakarta.batch.api.chunk.ItemReader, class jakarta.batch.api.chunk.AbstractItemReader, class org.acme.CustomItemReader]
	at io.quarkus.arc.impl.BeanManagerImpl.getReference(BeanManagerImpl.java:60)
	at io.quarkiverse.jberet.runtime.QuarkusBatchEnvironment$QuarkusArtifactFactory.create(QuarkusBatchEnvironment.java:97)
	at org.jberet.creation.ArtifactFactoryWrapper.create(ArtifactFactoryWrapper.java:39)
	at org.jberet.runtime.context.JobContextImpl.createArtifact(JobContextImpl.java:195)
	... 16 more
```
In the attached reproducer project [batch-provides.zip](https://github.com/quarkiverse/quarkus-jberet/files/11895956/batch-provides.zip):
* myJob / testRun uses `@Produces` factory method -> Job fails
* myJob2 / testRun2 uses bean level class -> Job succeeds

In QuarkusBatchEnvironment class
```
        @Override
        public Object create(String ref, Class<?> cls, ClassLoader classLoader) {
            BeanManager bm = Arc.container().beanManager();
            Bean<?> bean = bm.resolve(bm.getBeans(aliases.getOrDefault(ref, ref)));
            return bean == null ? null : bm.getReference(bean, bean.getBeanClass(), bm.createCreationalContext(bean));
        }
```
`bean.getBeanClass()` contains the factory class (MyJobProvider here), not the target bean class (CustomItemReader here).

--- a/core/deployment/src/test/java/io/quarkiverse/jberet/deployment/ChunkTest.java
+++ b/core/deployment/src/test/java/io/quarkiverse/jberet/deployment/ChunkTest.java
@@ -106,8 +106,15 @@ public class ChunkTest {
         }
     }
 
-    @Named
-    @Dependent
+    @ApplicationScoped
+    public static class PeopleProcessorProducer {
+        @Dependent
+        @Named
+        public PeopleProcessor peopleProcessor() {
+            return new PeopleProcessor();
+        }
+    }
+
     public static class PeopleProcessor implements ItemProcessor {
         @Override
         public Object processItem(Object item) {
@@ -115,7 +122,6 @@ public class ChunkTest {
             person.setPassword(UUID.randomUUID().toString());
             return person;
         }
-
     }
 
     @Dependent

--- a/core/runtime/src/main/java/io/quarkiverse/jberet/runtime/QuarkusBatchEnvironment.java
+++ b/core/runtime/src/main/java/io/quarkiverse/jberet/runtime/QuarkusBatchEnvironment.java
@@ -94,7 +94,7 @@ class QuarkusBatchEnvironment implements BatchEnvironment {
         public Object create(String ref, Class<?> cls, ClassLoader classLoader) {
             BeanManager bm = Arc.container().beanManager();
             Bean<?> bean = bm.resolve(bm.getBeans(aliases.getOrDefault(ref, ref)));
-            return bean == null ? null : bm.getReference(bean, bean.getBeanClass(), bm.createCreationalContext(bean));
+            return bean == null ? null : bm.getReference(bean, Object.class, bm.createCreationalContext(bean));
         }
 
         @Override

